pkg/virt-controller/watch/annotations.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package watch
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+func (c *VMIController) copyMultusAnnotationsFromPodToVmi(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	rateLimitedExecutor := c.sriovVmiCopyAnnotExecPool.LoadOrStore(vmi.UID)
+	err := rateLimitedExecutor.Exec(func() error {
```

Comment:
Hi @RamLavi, can you please elaborate on the need to use `rateLimitedExecutor`?
Isn't the VMI getting re-queued in case this function fails?

Reply:
good question. it's would give the same result but I am wondering whether we want to "err" the entire controller reconcile over this.. @EdDev what are you thoughts?